src/components/history/History.js

```diff
@@ -0,0 +1,66 @@
+import { HistoryCard } from "./HistoryCard";
+import { Header } from "../Header";
+import { SideNav } from "../SideNav";
+import { useReduce } from "../../providers/useReducerProvider";
+import { useEffect } from "react";
+import axios from "axios";
+import { useHistory } from "./HistoryContextProvider";
+import { useLoader } from "../home/LoaderContextProvider";
+import { Loader } from "../Loader";
+export const History = () => {
+  const { dispatch, setIsSideNav, state } = useReduce();
+  const { setHistoryData, historyData } = useHistory();
+  const { isLoader, setIsLoader } = useLoader();
+
+  useEffect(() => {
+    (async function () {
```

Comment:
If you are using `GET` request everywhere, you can create a function so that you don't have to repeat your code, you can also do it for `POST` request as well